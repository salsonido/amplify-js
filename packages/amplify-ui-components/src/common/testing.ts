import {
	boolean as booleanKnob,
	select as selectKnob,
	text as textKnob,
} from '@storybook/addon-knobs';

export const pixelThreshold = 10;

export const selectors = {
	// amplify-form-field
	formFieldDescription: '[data-test=form-field-description]',
};

export const knobs = {
	labelKnob(labelPlaceholder: string) {
		return textKnob('Label', labelPlaceholder);
	},

	descriptionKnob(descriptionPlaceholder: string) {
		return textKnob('Description', descriptionPlaceholder);
	},

	hintKnob(hintPlaceholder: string) {
		return textKnob('Hint', hintPlaceholder);
	},

	inputTypeKnob(inputTypePlaceholder: string) {
		return selectKnob(
			'Type',
			[
				'date',
				'email',
				'number',
				'password',
				'search',
				'tel',
				'text',
				'url',
				'time',
			],
			inputTypePlaceholder
		);
	},

	placeholderKnob(placeholderPlaceholder: string) {
		return textKnob('Placeholder', placeholderPlaceholder);
	},

	buttonTypeKnob(buttonTypePlaceholder: string) {
		return selectKnob(
			'Button type',
			['button', 'submit', 'reset'],
			buttonTypePlaceholder
		);
	},

	buttonTextKnob(buttonTextPlaceholder: string) {
		return textKnob('Button text', buttonTextPlaceholder);
	},

	toggleKnob(toggleDefault: boolean) {
		return booleanKnob('Toggle', toggleDefault);
	},

	disabledKnob(disabledDefault: boolean) {
		return booleanKnob('Disabled', disabledDefault);
	},

	formSectionHeaderKnob(headerPlaceholder: string) {
		return textKnob('Header text', headerPlaceholder);
	},

	formSectionSubmitKnob(submitPlaceholder: string) {
		return textKnob('Submit button text', submitPlaceholder);
	},

	tooltipAutoShowKnob(autoShowDefault: boolean) {
		return booleanKnob('Tooltip autoshow', autoShowDefault);
	},
};
