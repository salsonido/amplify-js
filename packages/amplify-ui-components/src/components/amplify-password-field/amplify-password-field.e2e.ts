import { newE2EPage } from '@stencil/core/testing';
import { pixelThreshold } from '../../common/testing';

describe('amplify-password-field e2e:', () => {
	it('should render a amplify-password-field', async () => {
		const page = await newE2EPage();
		await page.setContent(`<amplify-password-field></amplify-password-field>`);
		const el = await page.find('amplify-password-field');
		expect(el).not.toBeNull();

		const screenshot = await page.compareScreenshot('amplify-password-field', {
			fullPage: true,
		});
		expect(screenshot).toMatchScreenshot({
			allowableMismatchedPixels: pixelThreshold,
		});
	});
});
