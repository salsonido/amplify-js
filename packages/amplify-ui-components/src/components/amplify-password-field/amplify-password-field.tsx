import { I18n } from '@aws-amplify/core';
import { Component, Prop, FunctionalComponent, h } from '@stencil/core';
import { PASSWORD_SUFFIX } from '../../common/constants';
import { Translations } from '../../common/Translations';

@Component({
	tag: 'amplify-password-field',
})
export class AmplifyPasswordField {
	/** Based on the type of field e.g. sign in, sign up, forgot password, etc. */
	@Prop() fieldId: string = PASSWORD_SUFFIX;
	/** Used for the password label */
	@Prop() label: string = Translations.PASSWORD_LABEL;
	/** Used for the placeholder label */
	@Prop() placeholder: string = Translations.PASSWORD_PLACEHOLDER;
	/** Used as the hint in case you forgot your password, etc. */
	@Prop() hint: string | FunctionalComponent | null;
	/** The required flag in order to make an input required prior to submitting a form */
	@Prop() required: boolean = false;
	/** The callback, called when the input is modified by the user. */
	@Prop() handleInputChange?: (inputEvent: Event) => void;
	/** The value of the content inside of the input field */
	@Prop() value?: string;
	/** Attributes places on the input element: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes */
	@Prop() inputProps?: object;
	/** Will disable the input if set to true */
	@Prop() disabled?: boolean;

	render() {
		return (
			<amplify-form-field
				type="password"
				name="password"
				fieldId={this.fieldId}
				label={I18n.get(this.label)}
				placeholder={I18n.get(this.placeholder)}
				hint={this.hint}
				required={this.required}
				handleInputChange={this.handleInputChange}
				value={this.value}
				inputProps={this.inputProps}
				disabled={this.disabled}
			/>
		);
	}
}
