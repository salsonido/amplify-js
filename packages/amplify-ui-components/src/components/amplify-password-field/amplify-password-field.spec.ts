import { newSpecPage } from '@stencil/core/testing';
import { AmplifyPasswordField } from './amplify-password-field';

describe('amplify-password-field spec:', () => {
	describe('Component logic ->', () => {
		let passwordField;

		beforeEach(() => {
			passwordField = new AmplifyPasswordField();
		});

		it('should render hint to `undefined` by default', () => {
			expect(passwordField.hint).toBeUndefined();
		});

		it('should render placeholder to `Enter your password` by default', () => {
			expect(passwordField.placeholder).toEqual('Enter your password');
		});

		it('should render label to `Password *` by default', () => {
			expect(passwordField.label).toEqual('Password *');
		});

		it('should render fieldId to `password` by default', () => {
			expect(passwordField.fieldId).toEqual('password');
		});

		it('should render required to `false` by default', () => {
			expect(passwordField.required).toBe(false);
		});
	});
	describe('Render logic ->', () => {
		it('should render with `label` and `placeholder` values changed', async () => {
			const page = await newSpecPage({
				components: [AmplifyPasswordField],
				html: `<amplify-password-field label="New password *" placeholder="Enter new password"></amplify-password-field>`,
			});

			expect(page.root).toMatchSnapshot();
		});

		it('should render with default values', async () => {
			const page = await newSpecPage({
				components: [AmplifyPasswordField],
				html: `<amplify-password-field></amplify-password-field>`,
			});

			expect(page.root).toMatchSnapshot();
		});
	});
});
