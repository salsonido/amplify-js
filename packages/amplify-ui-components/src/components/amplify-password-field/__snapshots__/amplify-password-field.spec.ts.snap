// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-password-field spec: Render logic -> should render with \`label\` and \`placeholder\` values changed 1`] = `
<amplify-password-field label="New password *" placeholder="Enter new password">
  <amplify-form-field fieldid="password" label="New password *" name="password" placeholder="Enter new password" type="password"></amplify-form-field>
</amplify-password-field>
`;

exports[`amplify-password-field spec: Render logic -> should render with default values 1`] = `
<amplify-password-field>
  <amplify-form-field fieldid="password" label="Password *" name="password" placeholder="Enter your password" type="password"></amplify-form-field>
</amplify-password-field>
`;
